Add support for page fragments to Jakarta site

autumnfound opened the pull request:
Using new version of Hugo theme, a new templates section was added that allows the sites base page template to be retrieved programatically. These pages do not contain any page content, and will need to be added manually to be useful.

autumnfound committed:
Add support for page fragments to Jakarta site

Using new version of Hugo theme, a new templates section was added that allows the sites base page template to be retrieved programatically. These pages do not contain any page content, and will need to be added manually to be useful.

autumnfound commented:
This is updated to support the template pages changes I made in the base theme. I don't see any issues on a cursory glance through the site. Was there a specific feature you were worried might not work in the upgrade?

chrisguindon commented on package.json:
```diff
@@ -26,7 +26,7 @@
   },
   "dependencies": {
-    "eclipsefdn-hugo-solstice-theme": "0.0.181",
+    "eclipsefdn-hugo-solstice-theme": "0.0.186",
```
We should also update the Hugo version in the netlify file if we upgrade this package. See https://gitlab.eclipse.org/eclipsefdn/infrazilla/-/issues/1011 for more info.

We also need to update the readme.md when we make this change:
https://gitlab.eclipse.org/eclipsefdn/it/websites/eclipse.org/-/commit/f85703d8ff207a74cdc4ca0a3bd2e02549d52951

Finally, we had some issues with the spec project:

TLDR; yarn is not installed for that project. Can you also switch form yarn.lock to packages-lock.json with this change?

autumnfound committed:
Update netlify build to use NPM + v110 Hugo

autumnfound committed:
Fix missing polyfills in webpack build

autumnfound committed:
Remove unwanted build file from npm

chrisguindon commented on webpack.mix.js:
```diff
@@ -15,11 +15,20 @@ require('./node_modules/eclipsefdn-solstice-assets/webpack-solstice-assets.mix.j
 let mix = require('laravel-mix');
 mix.EclipseFdnSolsticeAssets();
 
+// required to resolve missing polyfills
```
If this is going to be needed for all the sites, let's fix it in solstice-assets so that all sites can benefit: 
https://gitlab.eclipse.org/eclipsefdn/it/webdev/solstice-assets/-/blob/main/webpack-solstice-assets.mix.js

autumnfound committed:
Update EFHST version and add semver vuln' patch

New EFHST version includes the changes for the webpack base config to provide the fallbacks for query string, meaning it is no longer required in this package.

Currently there is a compile time vulnerability with semver. An override was added to force it to the newest version as it should be stable even for old packages. Semver provides functionality to check NPM versions and is relatively stable otherwise.

autumnfound committed:
Set fixed version for EFHST

chrisguindon commented:
> This is updated to support the template pages changes I made in the base theme. I don't see any issues on a cursory glance through the site. Was there a specific feature you were worried might not work in the upgrade?

The specifications repo project. Once this is merged, we will need to upgrade the Hugo version and perhaps some build commands.

https://github.com/jakartaee/specifications/blob/master/netlify.toml
